src/lib/components/player/rank-chart.svelte

```diff
@@ -11,6 +11,15 @@
    });
 
    history.push(player.rank);
+   
+   // If this player stopped playing for a while, but has picked up again, then remove the 
+   // #999999 ranks at the beginning to make the chart more useful
+   if (history.some(h => h !== 999999)) {
+      while (history[0] === 999999) {
+         history.shift();
+      }
+   }
```

Comment:
This code is not clean.
First of all, you are using magic numbers. You should move the inactive rank in a const that you re-use later on, that way we can edit the rank if needed.
Secundo, you should not use while loops for this. There are plenty of ways to get rid of the inactive rank and the while way is just the worst one performance-wise.

What you could do is find the last index of the inactive rank and splice the array, that way we avoid unnecessary while loops and therefore, side effects in the future.

Reply:
Fair enough, thanks for the pointers!
I changed my fork (https://github.com/AaronC81/ScoreSaber-Frontend/commit/3d3241ed144535b1fcba10e7603a7329363a997e) but it doesn't seem to be updating here, presumably because it's closed. Can it be re-opened or should I open a new PR?

Reply:
Since your commits were merged, there is no way to re-open it I believe. Create a new one referencing this one and we should be good. 👍